scene/gui/control.cpp

```diff
@@ -2703,6 +2705,16 @@ void Control::get_argument_options(const StringName &p_function, int p_idx, List
 		}
 	}
 }
+
+String Control::get_configuration_warning() const {
+
+	if (data.mouse_filter == MOUSE_FILTER_IGNORE && data.tooltip != "") {
+		return TTR("The Hint Tooltip won't be displayed as the control's Mouse Filter is set to \"Ignore\". To solve this, set the Mouse Filter to \"Stop\" or \"Pass\".");
```

Comment:
I think you need to append that to `CanvasItem::get_configuration_warning()`, otherwise you would override any existing warning raised by a parent class (including `Node`'s which calls user defined methods). (See e.g. `scene/gui/range.cpp`).